rust/agama-software/src/callbacks/security.rs

```diff
@@ -0,0 +1,206 @@
+        let question = QuestionSpec::new(&text, "software.import_gpg")
+            .with_actions(&actions)
+            .with_data(&data)
+            .with_default_action("Skip");
+        let result = Handle::current()
+            .block_on(async move { ask_question(&self.questions, question).await });
+        let Ok(answer) = result else {
+            tracing::warn!("Failed to ask question {:?}", result);
+            return security::GpgKeyTrust::Reject;
+        };
+
+        answer
+            .action
+            .as_str()
+            .parse::<security::GpgKeyTrust>()
+            .unwrap_or(security::GpgKeyTrust::Reject)
+    }
+
+    fn unknown_key(&self, file: String, key_id: String, repository_alias: String) -> bool {
+        // TODO: localization for text when parameters in gextext will be solved
+        let text = if repository_alias.is_empty() {
+            format!(
+                "The file {file} is digitally signed with \
+                the following unknown GnuPG key: {key_id}. Use it anyway?"
+            )
+        } else {
+            format!(
+                "The file {file} from {repository_alias} is digitally signed with \
+                the following unknown GnuPG key: {key_id}. Use it anyway?"
+            )
+        };
+        let data = [("filename", file.as_str()), ("id", key_id.as_str())];
+        let question = QuestionSpec::new(&text, "software.unknown_gpg")
+            .with_yes_no_actions()
+            .with_data(&data);
+        let result = Handle::current()
+            .block_on(async move { ask_question(&self.questions, question).await });
+        let Ok(answer) = result else {
+            tracing::warn!("Failed to ask question {:?}", result);
+            return false;
+        };
+
+        answer.action == "Yes"
+    }
+
+    fn verification_failed(
+        &self,
+        file: String,
+        key_id: String,
+        key_name: String,
+        _key_fingerprint: String,
+        repository_alias: String,
+    ) -> bool {
+        // TODO: localization for text when parameters in gextext will be solved
+        let text = if repository_alias.is_empty() {
+            format!(
+                "The file {file} is digitally signed with the \
+                following GnuPG key, but the integrity check failed: {key_id} ({key_name}). \
+                Use it anyway?"
+            )
+        } else {
+            // TODO: Originally it uses repository url and not alias. Does it matter?
+            format!(
+                "The file {file} from {repository_alias} is digitally signed with the \
+                following GnuPG key, but the integrity check failed: {key_id} ({key_name}). \
+                Use it anyway?"
+            )
+        };
+        let data = [("filename", file.as_str())];
+        let question = QuestionSpec::new(&text, "software.verification_failed")
+            .with_yes_no_actions()
+            .with_data(&data);
+        let result = Handle::current()
```

Comment:
np: All functions are pretty much the same from this point. I wonder if we could reduce repetition a bit. But it can wait.